### PR TITLE
4616: Address db add

### PR DESCRIPTION
**Pull request by RuthBerry96**
Story: [AB#4616](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/4616)
Task: [AB#7212](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7212)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿CREATE TABLE [dbo].[Address]
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql:1]: Can you remove the square brackets from `[dbo]`?

(`[Address]` should have them.)

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿CREATE TABLE [dbo].[Address]
+(
+	AddressId INT IDENTITY(1,1) NOT NULL PRIMARY KEY,
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql]: Can you add a space after the comma in the `IDENTITY` function so it becomes `IDENTITY(1, 1)`? (Just slightly more readable.)

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/NHSD.BuyingCatalogue.Ordering.OrderingDatabase.sqlproj [hunk 3]
```diff
@@ -72,8 +72,9 @@
   <ItemGroup>
     <Build Include="Tables\OrderStatus.sql" />
     <Build Include="Tables\Order.sql" />
+    <Build Include="Tables\Address.sql" />
   </ItemGroup>
   <ItemGroup>
     <PostDeploy Include="Deployment\PostDeployment.sql" />
   </ItemGroup>
-</Project>
+</Project>
```
[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/NHSD.BuyingCatalogue.Ordering.OrderingDatabase.sqlproj:80]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql (hunk 2 above)
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql]: Can you move the definition of the primary key to a separate constraint so we can give it a name? Here's an example:

```sql
CONSTRAINT PK_Foobar PRIMARY KEY (AddressId)
```

The existing definition is absolutely fine functionally but the name of the primary key will be system generated (something like `PK__Address__091C2AFB8DEAA962`).

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    Line1 NVARCHAR(256) NULL, 
+    Line2 NVARCHAR(256) NULL, 
+    Line3 NVARCHAR(256) NULL, 
+    Line4 NVARCHAR(256) NULL, 
+    Line5 NVARCHAR(256) NULL, 
+    Town NVARCHAR(256) NULL, 
+    County NVARCHAR(256) NULL, 
+    Postcode NVARCHAR(10) NULL, 
+    Country NVARCHAR(256) NULL
+)
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql:13]: Just needs a terminating semicolon.

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+	AddressId INT IDENTITY(1,1) NOT NULL PRIMARY KEY, 
+    Line1 NVARCHAR(256) NULL, 
+    Line2 NVARCHAR(256) NULL, 
+    Line3 NVARCHAR(256) NULL, 
+    Line4 NVARCHAR(256) NULL, 
+    Line5 NVARCHAR(256) NULL, 
+    Town NVARCHAR(256) NULL, 
+    County NVARCHAR(256) NULL, 
+    Postcode NVARCHAR(10) NULL, 
+    Country NVARCHAR(256) NULL
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql]: Personally, I'd prefer lower case data types but I'm not sure we have any agreed standards on SQL formatting (yet) so your choice for now! 😃

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql (hunk 1 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql:1]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql (hunk 5 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql]: I'm sure we can discuss it on Friday 😄

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql (hunk 4 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql:13]: 👍

File: src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql (hunk 2 above)
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql]: 👍

[the remark this answers is not in the document]
Comment (RuthBerry96) [reply src/NHSD.BuyingCatalogue.Ordering.OrderingDatabase/Tables/Address.sql]: 👍

**Pull request merged**